install-template.sh

```diff
@@ -24,6 +24,34 @@ install_nix () {
   set -eu
 }
 
+test_cache_configured () {
+  nixConfig=$(nix --extra-experimental-features 'nix-command flakes' show-config) &&
+  (echo "$nixConfig" | grep --quiet "^substituters.*cache\.garnix\.io") &&
+  (echo "$nixConfig" | grep --quiet "^trusted-public-keys.*cache.garnix.io:CTFPyKSLcx5RMJKfLo5EEPUObbA78b0YQ2DTCJXqr9g=") &&
+  true
+}
+
+configure_cache () {
+  echo extra-substituters = https://cache.garnix.io | sudo tee -a /etc/nix/nix.conf > /dev/null
+  echo extra-trusted-public-keys = cache.garnix.io:CTFPyKSLcx5RMJKfLo5EEPUObbA78b0YQ2DTCJXqr9g= | sudo tee -a /etc/nix/nix.conf > /dev/null
+
+  echo trying to restart the nix-daemon...
+  # This ignores errors, for single-user installations.
+  if test "$(uname)" = "Linux" ; then
+    sudo systemctl restart nix-daemon.service || true
+  elif test "$(uname)" = "Darwin" ; then
+    sudo launchctl kickstart -k system/org.nixos.nix-daemon || true
+  else
+    echo "Unknown system: $(uname)"
+    echo Cancelling installation.
+    exit 1
+  fi
```

Comment:
I tried to detect whether a nix install is a multi user or single user installation. `grep nixbld /etc/group` works, but only on linux. I pushed [this branch](https://github.com/garnix-io/nix-tool-installer/compare/sh/cache...sh/check-for-multi-user?expand=1) that does attempt to do a portable check. But it's really a lot of complexity given that it doesn't have automated tests and is written in bash. So I would merge the version here. :shrug:

Reply:
I think “|| true” is good enough!